src/thor/costmatrix.cc

```diff
@@ -953,20 +959,25 @@ void CostMatrix::CheckReverseConnections(const uint32_t target,
         // to find for this source or target
         UpdateStatus(source, target);
       } else {
-        float oppcost = (fwd_predidx == kInvalidLabel) ? 0 : fwd_edgelabels[fwd_predidx].cost().cost;
-        float c = rev_pred.cost().cost + oppcost + fwd_label.transition_cost().cost;
+        // at this point, the found connection might still be somewhat trivial:
+        // the connecting edge might be an initial edge for either the given source or target
+        // there are two things we need to take care of in that case: we have to make sure to
+        // (1) only add the partial cost/distance for the connecting edge and (2) use the
+        // correct transition cost
+        auto& opp_label = fwd_predidx == kInvalidLabel ? fwd_label : fwd_edgelabels[fwd_predidx];
+        auto& label =
+            fwd_predidx == kInvalidLabel ? rev_edgelabels[rev_pred.predecessor()] : rev_pred;
+
+        Cost total_cost =
+            label.cost() + opp_label.cost() +
+            (fwd_predidx == kInvalidLabel ? rev_pred.transition_cost() : fwd_label.transition_cost());
```

Comment:
get the path up to the end of `label` from one direction and up to the end of `opp_label` from the other